### PR TITLE
DOCSP 33265 Resilience of mongosync

### PR DESCRIPTION
**Pull request by ltran-mdb2**
## DESCRIPTION

adding info about mongosync errors and resilience

## STAGING

https://preview-mongodbltranmdb2.gatsbyjs.io/cluster-sync/DOCSP-33265/reference/mongosync/#resilience
https://preview-mongodbltranmdb2.gatsbyjs.io/cluster-sync/DOCSP-33265/faq/#do-i-need-to-terminate-migration-early-if-i-see-error-or-failure-logs-

## JIRA

https://jira.mongodb.org/browse/DOCSP-33265

## BUILD LOG

https://workerpool-boxgs.mongodbstitch.com/pages/job.html?collName=queue&jobId=655cc0715b686d6bd97998c9

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Is this free of spelling errors?
- [x] Is this free of grammatical errors?
- [x] Is this free of staging / rendering issues?
- [x] Are all the links working?

## External Review Requirements

[What's expected of an external reviewer?](https://wiki.corp.mongodb.com/display/DE/Reviewing+Guidelines+for+the+MongoDB+Server+Documentation)

### CODE REVIEW

File: source/faq.txt [hunk 1]
```diff
@@ -115,3 +115,10 @@ If you see slow operation warnings, check CPU, memory, and network usage on the
 source and destination clusters.  If the clusters are underprovisioned for your
 needs, consider upgrading the cluster hardware.
 
+Do I need to terminate migration early if I see ``error`` or ``failure`` logs?
+------------------------------------------------------------------------------
+
+No, ``failure`` and ``error`` logs are non fatal errors and do not
+signal that you need to terminate ``mongosync`` early. These logs do not
+indicate that ``mongosync`` will fail or corrupt data. If a fatal error
+occurs, ``mongosync`` stops sync and writes a ``fatal`` log entry.
```
Comment (ianf-mongodb) [inline source/faq.txt:124]:
```suggestion
occurs, ``mongosync`` stops the sync and writes a fatal log entry. 
```

File: source/reference/mongosync.txt [hunk 2]
```diff
@@ -259,6 +259,16 @@ System Collections
 
 .. include:: /includes/collections/behavior-system-collections.rst
 
+Resilience
+~~~~~~~~~~
+
+``mongosync`` is resilient and able to handle non-fatal errors.
+``error`` and ``failure`` log entries do not indicate that ``mongosync``
```
Comment (ianf-mongodb) [inline source/reference/mongosync.txt:266]:
```suggestion
Error and failure log entries do not indicate that ``mongosync``
```

File: source/reference/mongosync.txt [hunk 3]
```diff
@@ -259,6 +259,16 @@ System Collections
...
+is failing or corrupting data. For example, if a network error occurs,
+``mongosync`` writes an ``error`` log but still completes sync. In the
+case that sync does not complete, ``mongosync`` writes a ``fatal`` log
```
Comment (ianf-mongodb) [inline source/reference/mongosync.txt:269]:
```suggestion
case that sync does not complete, ``mongosync`` writes a fatal log
```

File: source/reference/mongosync.txt [hunk 4]
```diff
@@ -259,6 +259,16 @@ System Collections
 
 .. include:: /includes/collections/behavior-system-collections.rst
 
+Resilience
+~~~~~~~~~~
+
+``mongosync`` is resilient and able to handle non-fatal errors.
+``error`` and ``failure`` log entries do not indicate that ``mongosync``
+is failing or corrupting data. For example, if a network error occurs,
+``mongosync`` writes an ``error`` log but still completes sync. In the
```
Comment (ianf-mongodb) [inline source/reference/mongosync.txt:268]:
```suggestion
``mongosync`` writes an error log but still completes sync. In the
```

**Review by ianf-mongodb — COMMENTED** [review]
Hi @ltran-mdb2, thanks for picking this up I left a few suggestions.

**Comment by ltran-mdb2** [comment]
Thanks for the feedback @ianf-mongodb ! Just completed your suggestions - could you take another look when you get a chance?

File: source/faq.txt [hunk 5]
```diff
@@ -115,3 +115,10 @@ If you see slow operation warnings, check CPU, memory, and network usage on the
 source and destination clusters.  If the clusters are underprovisioned for your
 needs, consider upgrading the cluster hardware.
 
+Should I stop a migration that encounters error or failure logs?
+----------------------------------------------------------------
+
+No, error and failure logs are non-fatal errors and do not signal that
+you need to terminate ``mongosync`` early. These logs do not indicate
```
Comment (ianf-mongodb) [inline source/faq.txt:122]: Update terminate -> stop per the [style guide](https://www.mongodb.com/docs/meta/style-guide/terminology/alphabetical-terms/#t)

```suggestion
you need to stop ``mongosync`` early. These logs do not indicate
```

File: source/reference/mongosync.txt [hunk 6]
```diff
@@ -259,6 +259,16 @@ System Collections
 
 .. include:: /includes/collections/behavior-system-collections.rst
 
+Resilience
+~~~~~~~~~~
+
+``mongosync`` is resilient and able to handle non-fatal errors.
+Error and failure log entries do not indicate that ``mongosync``
+is failing or corrupting data. For example, if a network error occurs,
+``mongosync`` writes an error log but still completes sync. In the
```
Comment (ianf-mongodb) [inline source/reference/mongosync.txt:268]:
```suggestion
``mongosync`` writes an error log but still completes the sync. In the
```

File: source/reference/mongosync.txt [hunk 7]
```diff
@@ -259,6 +259,16 @@ System Collections
...
+case that sync does not complete, ``mongosync`` writes a fatal log
```
Comment (ianf-mongodb) [inline source/reference/mongosync.txt:269]:
```suggestion
case that a sync does not complete, ``mongosync`` writes a fatal log
```

**Review by ianf-mongodb — APPROVED** [review]
Hi @ltran-mdb2 LGTM % 3 nits. Thanks for updates!

File: source/faq.txt (hunk 5 above)
Comment (ltran-mdb2) [reply source/faq.txt]: Done!

File: source/reference/mongosync.txt (hunk 6 above)
Comment (ltran-mdb2) [reply source/reference/mongosync.txt]: Done!

File: source/reference/mongosync.txt (hunk 7 above)
Comment (ltran-mdb2) [reply source/reference/mongosync.txt]: Done! Thanks for your help on this one!

**Comment by ltran-mdb2** [comment]
Hi @rohshar ! Could you give an external review to this PR on the resilience of mongosync?

File: source/faq.txt [hunk 8]
```diff
@@ -115,3 +115,10 @@ If you see slow operation warnings, check CPU, memory, and network usage on the
 source and destination clusters.  If the clusters are underprovisioned for your
 needs, consider upgrading the cluster hardware.
 
+Should I stop a migration that encounters error or failure logs?
```
Comment (rohshar) [inline source/faq.txt:118]:
```suggestion
Should I stop a migration that logs the word "error" or "failure"?
```

File: source/faq.txt [hunk 9]
```diff
@@ -115,3 +115,10 @@ If you see slow operation warnings, check CPU, memory, and network usage on the
...
+----------------------------------------------------------------
+
+No, error and failure logs are non-fatal errors and do not signal that
```
Comment (rohshar) [inline source/faq.txt:121]:
```suggestion
No, these logs are non-fatal errors and do not signal that
```

File: source/reference/mongosync.txt [hunk 10]
```diff
@@ -259,6 +259,15 @@ System Collections
 
 .. include:: /includes/collections/behavior-system-collections.rst
 
+Resilience
+~~~~~~~~~~
+
+``mongosync`` is resilient and able to handle non-fatal errors. Error
+and failure log entries do not indicate that ``mongosync`` is failing or
```
Comment (rohshar) [inline source/reference/mongosync.txt:266]:
```suggestion
``mongosync`` is resilient and able to handle non-fatal errors. Logs that
contain the word "error" or "failure" do not indicate that ``mongosync`` is failing or
```

**Review by rohshar — CHANGES_REQUESTED** [review]

File: source/faq.txt (hunk 9 above)
Comment (ltran-mdb2) [reply source/faq.txt]: Done!

File: source/reference/mongosync.txt (hunk 10 above)
Comment (ltran-mdb2) [reply source/reference/mongosync.txt]: Done!

**Comment by ltran-mdb2** [comment]
Thanks for the feedback @rohshar ! Could you take another look when you get a chance?

File: source/reference/mongosync.txt [hunk 11]
```diff
@@ -259,6 +259,16 @@ System Collections
 
 .. include:: /includes/collections/behavior-system-collections.rst
 
+Resilience
+~~~~~~~~~~
+
+``mongosync`` is resilient and able to handle non-fatal errors. Logs
+that contain the word "error" or "failure" do not indicate that
+``mongosync`` is failing or corrupting data. For example, if a network
+error occurs, ``mongosync`` writes an error log but still completes the
```
Comment (rohshar) [inline source/reference/mongosync.txt:268]:
```suggestion
error occurs, the ``mongosync`` logs may contain the word error, but mongosync is still able to complete the
```

File: source/faq.txt [hunk 12]
```diff
@@ -115,3 +115,11 @@ If you see slow operation warnings, check CPU, memory, and network usage on the
 source and destination clusters.  If the clusters are underprovisioned for your
 needs, consider upgrading the cluster hardware.
 
+Should I stop a migration that logs the word "error" or "failure"?
```
Comment (rohshar) [inline source/faq.txt:118]:
```suggestion
Should I stop a migration if I see the word "error" or "failure" in the logs?
```

**Review by rohshar — CHANGES_REQUESTED** [review]

File: source/reference/mongosync.txt (hunk 11 above)
Comment (ltran-mdb2) [reply source/reference/mongosync.txt:268]: Done!

**Review by rohshar — APPROVED** [review]

File: source/reference/mongosync.txt [hunk 13]
```diff
@@ -259,6 +259,16 @@ System Collections
...
+sync. In the case that a sync does not complete, ``mongosync`` writes a
+fatal log entry.
```
[the remark this answers is not in the document]
Comment (rohshar) [reply source/reference/mongosync.txt]: either is fine

**Pull request merged**